lib/cn/cn_metrics.h

```diff
@@ -6,155 +6,162 @@
- *   kvset_alen()  - sum of kblock and vblock alen
- *   kvset_wlen()  - sum of kblock and vblock wlen
- *   kvset_vulen() - sum of vblock ulen (n/a for kblocks)
+ * Sum of hblock, kblock and vblock alen
  */
-static inline u64
+static inline uint64_t
 kvset_alen(const struct kvset_stats *kst)
 {
     return kst->kst_halen + kst->kst_kalen + kst->kst_valen;
 }
 
-static inline u64
+/**
+ * Sum of hblock, kblock and vblock wlen
+ */
+static inline uint64_t
 kvset_wlen(const struct kvset_stats *kst)
 {
     return kst->kst_hwlen + kst->kst_kwlen + kst->kst_vwlen;
 }
 
-static inline u64
-kvset_vulen(const struct kvset_stats *kst)
-{
-    return kst->kst_vulen;
-}
-
 /**
- * struct cn_node_stats - node metrics used by compaction scheduler
- * @ns_hclen: estimated total hblock capacity (mpr_alloc_cap) after compaction
- * @ns_kclen: estimated total kblock capacity (mpr_alloc_cap) after compaction
- * @ns_vclen: estimated total vblock capacity (mpr_alloc_cap) after compaction
- * @ns_keys_uniq:  number of unique keys (estimated from hyperloglog)
- * @ns_pcap:      current size / max size as a percentage
- * @ns_kst:       sum of kvset_stats for all kvsets in node
+ * Node metrics used by compaction scheduler
  */
 struct cn_node_stats {
-    struct kvset_stats ns_kst;
-    u64                ns_keys_uniq;
-    u64                ns_hclen;
-    u64                ns_kclen;
-    u64                ns_vclen;
-    u16                ns_pcap;
+    struct kvset_stats ns_kst; //<! Sum of kvset stats for all kvsts in node
+    uint64_t ns_keys_uniq; //<! number of unique keys (estimated from HyperLogLog stats)
+    uint64_t ns_hclen;     //<! estimated total hblock capacity (mpr_alloc_cap) after compaction
+    uint64_t ns_kclen;     //<! estimated total kblock capacity (mpr_alloc_cap) after compaction
+    uint64_t ns_vclen;     //<! estimated total vblock capacity (mpr_alloc_cap) after compaction
+    uint16_t ns_pcap;      //<! current size / max size as a percentage (0 <= ns_pcap <= 100)
 };
 
 /**
- * Node derived stats:
- *   cn_ns_clen()  - estimated mpool capacity used by node after kv-compaction
- *   cn_ns_alen()  - current capacity used by node (sum of kvset alen)
- *   cn_ns_wlen()  - sum of kvset wlen
- *   cn_ns_vulen() - sum of kvset vblocks ulen
- *   cn_ns_keys()  - number of keys in kvset
- *   cn_ns_kblks() - number of kblocks in node
- *   cn_ns_vblks() - number of vblocks in node
+ * Sum of "alen" for all mblocks in node.
  */
-static inline u64
+static inline uint64_t
 cn_ns_alen(const struct cn_node_stats *ns)
 {
     return kvset_alen(&ns->ns_kst);
 }
 
-static inline u64
+/**
+ * Sum of "wlen" for all mblocks in node.
+ */
+static inline uint64_t
 cn_ns_wlen(const struct cn_node_stats *ns)
 {
     return kvset_wlen(&ns->ns_kst);
 }
 
-static inline u64
-cn_ns_vulen(const struct cn_node_stats *ns)
-{
-    return kvset_vulen(&ns->ns_kst);
-}
-
-static inline u64
+/**
+ * Estimated mpool capacity used by node after kv-compaction.
+ */
```

Comment:
Why use the doxygen comment leader when it's not a doxygen comment?

Reply:
Because it works for language servers, and there's no point in repeating the function name.  Do language servers without the `/**` format it nicely?

Reply:
I wouldn't know, what do you get when you generate all the doc output?